File: driver_app/lib/services/auth_service.dart
import 'api_client.dart';

class AuthService {
  AuthService(this._api);

  final ApiClient _api;

  Future<void> login(String email, String password) async {
    final response = await _api.postJson('/auth/login', {
      'email': email,
      'password': password,
    });
    final token = response['data']['access_token'] as String;
    await _api.saveToken(token);
  }

  Future<void> logout() => _api.clearToken();
}

File: frontend/src/services/authService.ts
import { apiGet, apiPost, clearTokens, setTokens } from "./apiClient";
import type { TokenResponse, UserProfile } from "../types/api";

export async function login(username: string, password: string): Promise<UserProfile> {
  const tokens = await apiPost<TokenResponse>("/auth/login", { username, password });
  setTokens(tokens.access_token, tokens.refresh_token);
  const profile = await apiGet<UserProfile>("/auth/me");
  if (profile.role === "DRIVER") {
    clearTokens();
    throw new Error("Drivers must use the mobile app.");
  }
  return profile;
}

export async function logout(): Promise<void> {
  const refreshToken = localStorage.getItem("prodrive_refresh_token");
  if (refreshToken) {
    try {
      await apiPost("/auth/logout", { refresh_token: refreshToken });
    } catch {
      // Ignore remote logout failures.
    }
  }
  clearTokens();
}

export async function fetchCurrentUser(): Promise<UserProfile | null> {
  if (!localStorage.getItem("prodrive_access_token")) {
    return null;
  }
  try {
    const profile = await apiGet<UserProfile>("/auth/me");
    if (profile.role === "DRIVER") {
      clearTokens();
      return null;
    }
    return profile;
  } catch {
    clearTokens();
    return null;
  }
}
